msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1263,6 +1268,131 @@ public boolean isSharedDevice() {
         return mPublicClientConfiguration.getIsSharedDevice();
     }
 
+    @Override
+    public String generateSignedHttpRequest(@NonNull final IAccount account,
+                                            @NonNull final PoPAuthenticationScheme popParameters) throws MsalException {
+        final ResultFuture<AsyncResult<GenerateShrResult>> future = new ResultFuture<>();
+
+        final GenerateShrCommand generateShrCommand = createGenerateShrCommand(
+                account,
+                popParameters,
+                new CommandCallback<GenerateShrResult, BaseException>() {
+                    @Override
+                    public void onCancel() {
+                        // Not cancellable
+                    }
+
+                    @Override
+                    public void onError(@NonNull final BaseException error) {
+                        future.setResult(
+                                new AsyncResult<GenerateShrResult>(
+                                        null,
+                                        baseExceptionToMsalException(error)
+                                )
+                        );
+                    }
+
+                    @Override
+                    public void onTaskCompleted(@NonNull final GenerateShrResult generateShrResult) {
+                        future.setResult(new AsyncResult<>(generateShrResult, null));
+                    }
+                },
+                "1100" // Public API ID
```

Comment:
move this to `PublicApiId.java`

Reply:
cc33739f1c87ea2ae37c86ab5a15ef440c3a5b6e